src/lua/compat.lua

```diff
@@ -0,0 +1,339 @@
+-- compat.lua -- internal module intended to solve compatibility problems in
+-- different parts of Tarantool. Introduced in gh-7000, see also gh-6912.
+
+local NEW = true
+local OLD = false
+
+-- luacheck: no unused
+local internal = require('internal.compat')
+
+local options_format = {
+    default        = 'string',
+    brief          = 'string',
+    obsolete       = 'string/nil',
+    run_action_now = 'boolean/nil',
+    action         = 'function/nil',
```

Comment:
Current implementation forbids to change an option without an action:

```yaml
# tarantool> compat.add_option({name = 'foo', default = 'old', brief = 'xxx'})
# tarantool> compat.foo = 'new'
---
- error: 'builtin/compat.lua:125: attempt to call field ''action'' (a nil value)'
...
```

Reply:
Fixed.

```diff
+     if option.obsolete and option.action == nil then
+        error(("Invalid option table for %s, bad action (function is expected)"):format(name))
+     end

Reply:
Now it fails earlier:

```yaml
# tarantool> compat = require('tarantool').compat
# tarantool> compat.add_option({name = 'foo', default = 'old', brief = 'xxx'})
---
- error: 'builtin/compat.lua:98: Invalid option table for foo, bad action (function
    is expected)'
...
```

Why I can't add an option without an action to use it like `if compat.foo.current == 'new' then <..new behaviour..> else <..old behavior..> end`?

Reply:
We didn't intend such usage, I think they can be easily achieved with current version (e.g. by providing empty action func). But on the other hand, if we weaken option_def rules,  it may result in more mistakes and tipos.

Reply:
I would prefer optional `action`, but okay.